frontend/public/module/k8s/k8s.ts

```diff
@@ -19,9 +19,11 @@ export const referenceFor = (obj: K8sResourceKind): GroupVersionKind => obj.kind
   ? `${groupVersionFor(obj.apiVersion).group}:${groupVersionFor(obj.apiVersion).version}:${obj.kind}`
   : '';
 
-export const referenceForCRD = (obj: CustomResourceDefinitionKind): GroupVersionKind => (
-  `${obj.spec.group}:${obj.spec.version}:${obj.spec.names.kind}`
-);
+export const referenceForCRD = (obj) => {
+  return obj.apiGroup
+    ? `${obj.apiGroup}:${obj.apiVersion}:${obj.kind}`
+    : `${obj.spec.group}:${obj.spec.version}:${obj.spec.names.kind}`;
```

Comment:
These should be two different functions, one that takes in an instance and one that takes in the CRD.